### ISSUE TITLE
superagi-gui-1: Invalid next.config.js options detected... The root value has an unexpected property, webpackDevMiddleware

### ISSUE THREAD

**Issue by matthewcorven** [TransformerOptimus/SuperAGI#250]
Right out of the box, upon startup of the docker image, I am getting warnings from superagi-gui-1:

```sh
2023-06-07 14:26:29 
2023-06-07 14:26:29 > super-agi@0.1.0 dev
2023-06-07 14:26:29 > next dev
2023-06-07 14:26:29 
2023-06-07 14:26:30 - ready started server on 0.0.0.0:3000, url: http://localhost:3000
2023-06-07 14:27:20 Attention: Next.js now collects completely anonymous telemetry regarding usage.
2023-06-07 14:27:20 This information is used to shape Next.js' roadmap and prioritize features.
2023-06-07 14:27:20 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
2023-06-07 14:27:20 https://nextjs.org/telemetry
2023-06-07 14:27:20 
2023-06-07 14:29:10 - event compiled client and server successfully in 104.5s (781 modules)
2023-06-07 14:29:22 - wait compiling...
2023-06-07 14:29:27 - event compiled client and server successfully in 4.1s (781 modules)
2023-06-07 18:14:29 
2023-06-07 18:14:29 > super-agi@0.1.0 dev
2023-06-07 18:14:29 > next dev
2023-06-07 18:14:29 
2023-06-07 18:14:30 - ready started server on 0.0.0.0:3000, url: http://localhost:3000
2023-06-07 18:14:39 - event compiled client and server successfully in 3.7s (781 modules)
2023-06-07 18:14:43 - wait compiling...
2023-06-07 18:14:44 - event compiled client and server successfully in 1109 ms (781 modules)
2023-06-07 18:14:50 - wait compiling /_error (client and server)...
2023-06-07 18:14:51 - event compiled client and server successfully in 854 ms (782 modules)
2023-06-07 18:15:40 - wait compiling /favicon.ico/route (client and server)...
2023-06-07 18:15:42 - event compiled successfully in 2.3s (135 modules)
2023-06-07 23:00:36 
2023-06-07 23:00:36 > super-agi@0.1.0 dev
2023-06-07 23:00:36 > next dev
2023-06-07 23:00:36 
2023-06-07 23:00:36 - ready started server on 0.0.0.0:3000, url: http://localhost:3000
2023-06-07 23:00:43 - event compiled client and server successfully in 3.3s (781 modules)
2023-06-07 23:00:45 - wait compiling...
2023-06-07 23:00:46 - event compiled client and server successfully in 1071 ms (781 modules)
2023-06-07 23:00:47 - wait compiling /_error (client and server)...
2023-06-07 23:00:48 - event compiled client and server successfully in 427 ms (782 modules)
2023-06-07 14:26:52 - warn Invalid next.config.js options detected: 
2023-06-07 14:26:52 - warn     The root value has an unexpected property, webpackDevMiddleware, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, configOrigin, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, modularizeImports, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, skipMiddlewareUrlNormalize, skipTrailingSlashRedirect, staticPageGenerationTimeout, swcMinify, target, trailingSlash, transpilePackages, typescript, useFileSystemPublicRoutes, webpack).
2023-06-07 14:26:52 - warn See more info here: https://nextjs.org/docs/messages/invalid-next-config
```

**Comment by hanhyalex123** [comment]
same, there is something wrong whit websocket

**Comment by hanhyalex123** [comment]
i have solved, if you are connecting to remote server, you should type exceat ip address of you server at config.yaml

**Comment by Panczo** [comment]
I have the same problem, I tried to adjust next.config.js as below but it stuck on Initializing SuperAGI... 

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
    assetPrefix: process.env.NODE_ENV === "production" ? "/" : "./",
    // webpackDevMiddleware: config => {
    // config.watchOptions = {
    //   poll: 1000,
    //   aggregateTimeout: 300,
    // }
    // return config
  // }
}

module.exports = nextConfig

``` 


Additionally I see another error:

```bash
| - wait compiling /_error (client and server)...
superagi-gui-1              | - event compiled client and server successfully in 390 ms (782 modules)
superagi-gui-1              | Do not add stylesheets using next/head (see <link rel="stylesheet"> tag with href="https://fonts.googleapis.com/css2?family=Source+Code+Pro&display=swap"). Use Document instead. 
```

**Comment by GoBooth** [comment]
I'm getting the same problem.

after I run docker-compose up --build, all superagi containers run, except for superagi-gui-1.
when I manually run it, this is the error I get:

superagi-gui-1              | 
superagi-gui-1              | > super-agi@0.1.0 dev
superagi-gui-1              | > next dev
superagi-gui-1              | 
superagi-gui-1              | - ready started server on 0.0.0.0:3000, url: http://localhost:3000
superagi-gui-1              | Segmentation fault
superagi-gui-1 exited with code 139
superagi-gui-1              | 
superagi-gui-1              | > super-agi@0.1.0 dev
superagi-gui-1              | > next dev
superagi-gui-1              | 
superagi-gui-1              | - ready started server on 0.0.0.0:3000, url: http://localhost:3000
superagi-gui-1              | - warn Invalid next.config.js options detected: 
superagi-gui-1              | - warn     The root value has an unexpected property, **webpackDevMiddleware**, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, configOrigin, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, modularizeImports, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, skipMiddlewareUrlNormalize, skipTrailingSlashRedirect, staticPageGenerationTimeout, swcMinify, target, trailingSlash, transpilePackages, typescript, useFileSystemPublicRoutes, webpack).
superagi-gui-1              | - warn See more info here: https://nextjs.org/docs/messages/invalid-next-config
superagi-gui-1 exited with code 1

**Comment by GoBooth** [comment]
> I have the same problem, I tried to adjust next.config.js as below but it stuck on Initializing SuperAGI...
> 
> ```js
> /** @type {import('next').NextConfig} */
> const nextConfig = {
>     assetPrefix: process.env.NODE_ENV === "production" ? "/" : "./",
>     // webpackDevMiddleware: config => {
>     // config.watchOptions = {
>     //   poll: 1000,
>     //   aggregateTimeout: 300,
>     // }
>     // return config
>   // }
> }
> 
> module.exports = nextConfig
> ```
> 
> Additionally I see another error:
> 
> ```shell
> | - wait compiling /_error (client and server)...
> superagi-gui-1              | - event compiled client and server successfully in 390 ms (782 modules)
> superagi-gui-1              | Do not add stylesheets using next/head (see <link rel="stylesheet"> tag with href="https://fonts.googleapis.com/css2?family=Source+Code+Pro&display=swap"). Use Document instead. 
> ```

same.

**Comment by neelayan7** [comment]
@matthewcorven is this issue still persisting with the latest version releases?

**Comment by GoBooth** [comment]
I deleted the entire cloned folder, and the docker container, I did a new git clone, and compose build. 

-gui still wont start.

however, I no longer ger any error messages.

`superagi-gui-1              | 
superagi-gui-1              | > super-agi@0.1.0 dev
superagi-gui-1              | > next dev
superagi-gui-1              | 
superagi-gui-1              | - ready started server on 0.0.0.0:3000, url: http://localhost:3000
superagi-gui-1 exited with code 1`
